lib/friends/introvert.rb

```diff
@@ -777,5 +779,52 @@ def thing_with_name_in(type, text)
     def bad_line(expected, line_num)
       raise FriendsError, "Expected \"#{expected}\" on line #{line_num}"
     end
+
+    def decide_default_location_output(activity)
+      @activities = stable_sort(@activities)
```

Comment:
I think you mentioned before that we can use `stable_sort` in case the user somehow manually edits the activities in the `friends.md` file out of order. And even though this is an edge case, I think it's worth taking a precaution against, only because order is so important to the functionality in this PR.

And actually, I now think we should try and include `stable_sort` wherever we have functionality that depends on the order of activities, because it is that important. For example, we could also include it in the [set_implicit_locations!](https://github.com/shen-sat/friends/blob/a7794d507302226a950c407c99feaa153cb63e3b/lib/friends/introvert.rb#L663) method.

And following this thinking even further, maybe we should just add the `stable_sort` when we read the file? That way we don't have to repeat `stable_sort` in different places in the code.